README.md

```diff
@@ -1,23 +1,39 @@
-
 # Blueprint
 
-### Declarative UI construction for iOS, written in Swift
+[![Tests](https://github.com/square/Blueprint/actions/workflows/tests.yaml/badge.svg)](https://github.com/square/Blueprint/actions/workflows/tests.yaml)
+[![Generate and publish docs](https://github.com/square/Blueprint/actions/workflows/docs.yml/badge.svg)](https://github.com/square/Blueprint/actions/workflows/docs.yml)
+[![Linter](https://github.com/square/Blueprint/actions/workflows/lint.yml/badge.svg)](https://github.com/square/Blueprint/actions/workflows/lint.yml)
 
-Blueprint greatly simplifies the task of building and updating views as application state changes.
+## Declarative UI construction for iOS, written in Swift
 
-*We still consider Blueprint experimental (and subject to major breaking API changes), but it has been used within Square's production iOS apps.*
+Blueprint greatly simplifies the task of building and updating views as application state changes.
 
 ```swift
-let rootElement = Label(text: "Hello from Blueprint!")
-let view = BlueprintView(element: rootElement)
+let view = BlueprintView(element: Label(text: "Hello from Blueprint!"))
 ```
 
-Generated documentation is available at [square.github.io/Blueprint](https://square.github.io/Blueprint/)
+### What does this library do?
 
-### Getting Started
+Blueprint provides an architecture that allows you to:
+- Declaratively define a UI hierarchy as pure values (Swift structs and enums).
+- Display that hierarchy within your application.
+- Update that hierarchy as application state changes (including animated transitions).
+
+### When should I use it?
+
+Use Blueprint any time you want to display a view hierarchy, but don't want to manage view lifecycle (hint: managing view lifecycle is a large portion of most conventional UIKit code). There are times when you *want* to manage view lifecycle (complex animations and transitions are a good example), and for these cases you may want to stick with a conventional approach.
+
+### How does it interact with UIKit?
+
+Blueprint is not a replacement for UIKit! From the beginning, Blueprint has been designed as a compliment to all of the powerful tools that come with the platform. You can use Blueprint to manage the display of a single view controller, or of a single view representing a small part of the screen. Likewise, it's straightforward to host standard views and controls *within* a blueprint hierarchy, always leaving you with an escape hatch.
+
+### How does it interact with SwiftUI?
+
+They serve similar purposes, and SwiftUI has heavily influenced Blueprint's API. However, Blueprint predates SwiftUI, and works a bit differently. You can host a `BlueprintView` in SwiftUI to embed Blueprint within SwiftUI, and you can also use the provided `ElementView` to create Xcode previews of Blueprint. Because SwiftUI is hosted by a UIViewController, and Blueprint is hosted in a UIView, we don't provide a way to embed SwiftUI within Blueprint.
```

Comment:
I cleaned this file up a lot, but it's mostly moving and deleting. This paragraph is new.